src/mmds/src/lib.rs

```diff
@@ -197,11 +198,60 @@ fn respond_to_get_request_unchecked(request: Request) -> Response {
     }
 }
 
+fn respond_to_put_request(request: Request, custom_headers: CustomHeaders) -> Response {
+    // Get token lifetime value.
+    let ttl_seconds = match custom_headers.x_aws_metadata_token_ttl_seconds() {
+        Some(ttl_seconds) => ttl_seconds,
+        None => {
+            let error_msg = "Token time to live value not found.".to_string();
```

Comment:
Let's validate token TTL _after_ validating token path. That way the errors are more helpful.

Reply:
Done